src/proofPointAuthenticator.ts

```diff
@@ -0,0 +1,128 @@
+import { EthereumAddressResolver } from "./ethereumAddressResolver";
+import {
+  EthereumProofPointRegistry,
+  ETHEREUM_PROOF_TYPE,
+} from "./ethereumProofPointRegistry";
+import { ProofPoint } from "./proofPoint";
+import { ProofPointId, ProofPointIdType } from "./proofPointId";
+import { ProofPointStatus } from "./proofPointStatus";
+import { ProofPointValidateResult } from "./proofPointValidateResult";
+
+/**
+ * Proof point authenticator
+ * An object capable of authenticating a proof point.
+ */
+export interface ProofPointAuthenticator {
+  /**
+   * Determines whether the given proof point is authentic.
+   * @param id The ID of the proof point to check.
+   * @param proofPoint The proof point to check.
+   * @returns true if the proof point is authentic.
+   */
+  authenticate(
+    id: ProofPointId,
+    proofPoint: ProofPoint
+  ): Promise<ProofPointValidateResult>;
+}
+
+/**
+ * General proof point authenticator
+ * A @ProofPointAuthenticator capable of handling Ethereum based proof points and Web based proof points.
+ */
+export class GeneralProofPointAuthenticator {
```

Comment:
Not sure I see the purpose of this class currently?

Reply:
Valid. Removed.